seqr/views/apis/dataset_api_tests.py

```diff
@@ -69,11 +72,12 @@ def test_add_variants_dataset(self, mock_random):
         }}, method=urllib3_responses.POST)
         MOCK_FILE_ITER.return_value = StringIO('NA19678_1,NA19678\n')
 
-        response = self.client.post(url, content_type='application/json', data=json.dumps({
-            'elasticsearchIndex': INDEX_NAME,
-            'mappingFilePath': 'mapping.csv',
-            'datasetType': 'VARIANTS',
-        }))
+        with mock.patch('seqr.views.utils.permissions_utils.ANALYST_PROJECT_CATEGORY', 'analyst-projects'):
```

Comment:
you should patch the whole function or none of it. We have "Non-Analyst Project" as a test fixture thats not an analyst project that you can use for testing the non analyst project case, or if that really doesn't work for some reason do a second test function to test the analyst notification on its own

Reply:
I found family 14 and individual 18 belong to the "Non-Analyst Project" in the fixture data. I will use these data to test the non-analyst project case.